### PR TITLE
Ep 2124 updated search

### PR DESCRIPTION
**Pull request by ancashoria**
### Update search to also work on the location details page

- Refactored <Sidebar/> component to have clear responsibility and make it more reusable
- Refactored <Places /> component
- Added full search functionality on the location details page

### Checklist

- [x] I have added documentation for new/changed functionality in this PR or in /docs
- [x] All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not `master`

### CODE REVIEW

File: packages/earth-map/src/components/org-switcher/component.tsx [hunk 1]
```diff
@@ -1,11 +1,14 @@
 import React, { useContext, useState } from 'react';
-import { Auth0Context } from 'auth/auth0';
+import { remove } from 'lodash';
+import classNames from 'classnames';
```
Comment (corinacioloca) [inline packages/earth-map/src/components/org-switcher/component.tsx:3]: don't think logo should be in org-switcher, either rename the component in something more generic (eg header) or extract if from here

File: packages/earth-map/src/components/sidebar/component.tsx [hunk 2]
```diff
@@ -44,32 +76,32 @@ class SidebarComponent extends React.Component<ISidebar> {
...
-            <button type="button" onClick={() => this.onClickItem()} className="sidebar--btn">
-              <i
-                className={classNames({
-                  'ng-body-color ng-text-bold ng-icon-small': true,
-                  'ng-icon-directionleft': open,
-                  'ng-icon-directionright': !open,
-                })}
-              />
-            </button>
+            <SidebarClose
```
Comment (corinacioloca) [inline packages/earth-map/src/components/sidebar/component.tsx]: SidebarToggleButton? cause if opens/closes sidebar

File: packages/earth-map/src/components/sidebar/sidebar-layout-search/component.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            }}
+          >
+            <Layers />
+          </animated.div>
+        )}
+      </LayersDropdown>
+      <div style={{ height: '100%', overflow: 'auto' }}>
+        <div style={{ position: 'sticky', top: 0, zIndex: 1 }}>
+          <OrgSwitcher />
+          <SearchBox showClose={showX} />
```
Comment (corinacioloca) [inline packages/earth-map/src/components/sidebar/sidebar-layout-search/component.tsx:76]: showClose is not declared on SearchBox props interface

File: packages/earth-map/src/components/sidebar/styles.scss [hunk 4]
```diff
@@ -1,18 +1,44 @@
...
+    > svg {
+      display: block;
+    }
+  }
+  &.expanded {
+    width: 500px;
+  }
+}
+
+.ng-ep-kicker {
```
Comment (corinacioloca) [inline packages/earth-map/src/components/sidebar/styles.scss]: ng-ep-kicker is not in this component, please move it where the element is used

File: packages/earth-map/src/pages/earth/component.tsx [hunk 5]
```diff
@@ -52,27 +33,9 @@ class EarthPage extends React.Component<IEarth> {
...
-                  <Layers />
-                </animated.div>
-              )}
-            </LayersDropdown>
-            <Places />
-            {!!selected && <IndexSidebar />}
-          </SidebarPanel>
-        </div>
+        <Sidebar>
+          <SidebarLayoutSearch selected={!!selected} />
```
Comment (corinacioloca) [inline packages/earth-map/src/pages/earth/component.tsx:37]: can we just have Sidebar here? and not use children in the sidebar component? don't see why we need this abstraction

Comment (ancashoria) [reply packages/earth-map/src/pages/earth/component.tsx:37]: I wanted the `<Sidebar />` to be solely responsible for it's layout and the toggle functionality, not for the content inside it. In this way we could put anything inside the sidebar on different pages if that's needed in the future.

File: packages/earth-map/src/components/org-switcher/component.tsx (hunk 1 above)
Comment (ancashoria) [reply packages/earth-map/src/components/org-switcher/component.tsx:3]: Will rename it to `<Header/>`

File: packages/earth-map/src/components/sidebar/component.tsx (hunk 2 above)
Comment (ancashoria) [reply packages/earth-map/src/components/sidebar/component.tsx]: Will rename it to `<SidebarToggle/>`

**Review by corinacioloca — APPROVED** [review]

**Pull request merged**